test/Npgsql.Tests/ConnectionTests.cs

```diff
@@ -109,12 +109,21 @@ await using (var killingConn = await OpenConnectionAsync())
             Assert.That(conn.FullState, Is.EqualTo(ConnectionState.Broken));
             Assert.That(conn.State, Is.EqualTo(ConnectionState.Closed));
             Assert.That(eventClosed, Is.True);
+            Assert.That(conn.Connector is null);
+            Assert.AreEqual(0, conn.Pool.Statistics.Total);
 
-            await conn.CloseAsync();
+            if (openFromClose)
+            {
+                await conn.CloseAsync();
 
-            Assert.That(conn.State, Is.EqualTo(ConnectionState.Closed));
-            Assert.That(conn.FullState, Is.EqualTo(ConnectionState.Closed));
-            Assert.That(eventClosed, Is.True);
+                Assert.That(conn.State, Is.EqualTo(ConnectionState.Closed));
+                Assert.That(conn.FullState, Is.EqualTo(ConnectionState.Closed));
+                Assert.That(eventClosed, Is.True);
+            }
+
+            Assert.DoesNotThrowAsync(conn.OpenAsync);
```

Comment:
Thanks, this is good coverage. Maybe also execute `SELECT 1` to make sure the connection really works.